backend/src/main/java/eu/clarin/switchboard/resources/DataResource.java

```diff
@@ -129,10 +126,59 @@ public Response getFile(String idString, String mediatype) throws Throwable {
                 mediatype = mediatype + ";charset=utf-8";
             }
         }
-        return Response.ok(fileStream)
-                .type(mediatype)
-                .header("content-disposition", "attachment; filename=" + fi.getFilename())
-                .build();
+
+        if (range == null || range.isEmpty()) {
+            StreamingOutput fileStream = output -> {
+                byte[] data = Files.readAllBytes(fi.getPath());
+                output.write(data);
+                output.flush();
+            };
+            return Response.ok(fileStream)
+                    .type(mediatype)
+                    .header("content-disposition", "attachment; filename=" + fi.getFilename())
+                    .build();
+        }
+
+        String[] ranges = range.split("=");
+        if (ranges.length < 2) {
+            return Response.status(Response.Status.BAD_REQUEST).entity("bad range header").build();
+        }
+        ranges = ranges[1].split("-");
+        if (ranges.length > 2) {
+            return Response.status(Response.Status.BAD_REQUEST).entity("only one range is acceptable").build();
+        }
+
+        final int start = Integer.parseInt(ranges[0]);
+
+        // Chunk media if the range upper bound is unspecified. Chrome sends "bytes=0-"
+        final int requestedEnd = ranges.length < 2 ? start + MORE_CONTENT_BUFFER_SIZE : Integer.parseInt(ranges[1]);
+
+        final RandomAccessFile file = new RandomAccessFile(fi.getPath().toFile(), "r");
+        file.seek(start);
+
+        final int end = requestedEnd < file.length() ? requestedEnd : (int) (file.length() - 1);
+
+        StreamingOutput fileStream = output -> {
+            final byte[] buf = new byte[MORE_CONTENT_BUFFER_SIZE];
+            int len = end - start + 1;
+            try {
+                while (len != 0) {
+                    int read = file.read(buf, 0, Math.min(buf.length, len));
+                    output.write(buf, 0, read);
+                    len -= read;
+                }
+            } finally {
+                file.close();
+            }
+            output.flush();
+        };
```

Comment:
This really needs some comments - at least one line on top to state what it's doing. Or even better replace with something from a library or else put it in a method with an explanatory name :D